Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func prepareForReuse() {
+        self.titleText.font = DynamicFontHelper.defaultHelper.DefaultSmallFontBold
+    }
+    
+    func closeTab() {
+        delegate?.tabCellDidClose(self)
+    }
+    
+    override func drawRect(rect: CGRect) {
+        super.drawRect(rect)
+        guard seperatorLine else {
+            return
+        }
+        let context = UIGraphicsGetCurrentContext();
+        CGContextSaveGState(context)
+        CGContextSetLineCap(context, CGLineCap.Square)
+        CGContextSetStrokeColorWithColor(context, UIColor.whiteColor().colorWithAlphaComponent(0.2).CGColor)
+        CGContextSetLineWidth(context, 1.0)
+        CGContextMoveToPoint(context, 0, TopTabsUX.BackgroundSeparatorLinePadding)
+        CGContextAddLineToPoint(context, 0, frame.size.height-TopTabsUX.BackgroundSeparatorLinePadding)
+        CGContextStrokePath(context)
+        CGContextRestoreGState(context)
+    }
+}
+
+class BezierView: UIView {
+    var fillColor: UIColor?
+    init() {
+        super.init(frame: CGRect.zero)
+        self.backgroundColor = UIColor.clearColor()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func drawRect(rect: CGRect) {
+        super.drawRect(rect)
+        
+        guard let fillColor = self.fillColor else {
+            return
+        }
+        
+        let width = frame.width
+        let height = frame.height
+        let x1: CGFloat = 32.84
+        let x2: CGFloat = 5.1
+        let x3: CGFloat = 19.76
+        let x4: CGFloat = 58.27
+        let x5: CGFloat = -12.15
+        
+        //// Bezier Drawing
+        let bezierPath = UIBezierPath()
+        bezierPath.moveToPoint(CGPoint(x: width, y: height))
+        bezierPath.addCurveToPoint(CGPoint(x: width-x1, y: 0), controlPoint1: CGPoint(x: width-x3, y: height), controlPoint2: CGPoint(x: width-x2, y: 0))
+        bezierPath.addCurveToPoint(CGPoint(x: x1, y: 0), controlPoint1: CGPoint(x: width-x4, y: 0), controlPoint2: CGPoint(x: x4, y: 0))
+        bezierPath.addCurveToPoint(CGPoint(x: 0, y: height), controlPoint1: CGPoint(x: x2, y: 0), controlPoint2: CGPoint(x: x3, y: height))
+        bezierPath.addCurveToPoint(CGPoint(x: width, y: height), controlPoint1: CGPoint(x: x5, y: height), controlPoint2: CGPoint(x: width-x5, y: height))
+        bezierPath.closePath()
+        bezierPath.miterLimit = 4;
+        
+        fillColor.setFill()
+        bezierPath.fill()
+    }
+}
+
+extension TopTabsViewController : WKNavigationDelegate {
+    func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        collectionView.reloadData()
+    }
+    
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
```

Comment:
Does this get called anywhere? Just trying to figure out the motivation for implementing this delegate.